src/wazuh_modules/wm_sca.c

```diff
@@ -625,6 +629,27 @@ static int wm_sca_check_policy(cJSON *policy, cJSON *profiles) {
             read_id = (int *) realloc(read_id, sizeof(int) * (i + 2));
             read_id[i] = check_id->valueint;
             read_id[i + 1] = 0;
+
+            rules_id = cJSON_GetObjectItem(check, "rules");
+
+            cJSON_ArrayForEach(rule, rules_id){
+                if (rules_id == NULL) {
+                    merror("Rules not found.");
+                    free(read_id);
+                    return retval;
+                }
+
+                rules_n++;
+
+                if (rules_n > 255) {
+                    retval = 1;
+                    free(read_id);
+                    mwarn("Policy check with more than 255 rules. Ignoring it.");
```

Comment:
Why this is a warning and the previous one an error? Change it to error. It also can include more info: `merror("Invalid check %d: Maximum number of rules is 255", check_id->valueint);`